hoptimator-util/src/main/java/com/linkedin/hoptimator/util/planner/PipelineRules.java

```diff
@@ -134,7 +144,14 @@ public RelNode copy(RelTraitSet traitSet, List<RelNode> inputs) {
 
     @Override
     public void implement(Implementor implementor) throws SQLException {
-      implementor.setSink(database, table.getQualifiedName(), table.getRowType(), Collections.emptyMap());
+      RelDataTypeFactory typeFactory = new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+      RelDataType flattened = DataTypeUtils.flatten(table.getRowType(), typeFactory);
```

Comment:
@jogrogan I wonder if this would make `KEY ROW(...)` possible? If you were to take a Venice key schema and shove it into a nested record named `KEY`, it would get flattened as `KEY$FOO, KEY$BAR` automatically. And then in the sink table it would appear as `KEY_FOO, KEY_BAR` (see comment below).

Reply:
I wasn't thinking of keys when I wrote this, but it seems like it will help. We could have `select key$foo from venice.t1`, and the result would be `key_foo` in the physical venice sink.

Reply:
Yea that makes sense, I can play around with it once merged